parlai/scripts/generate_model_card.py

```diff
@@ -0,0 +1,1485 @@
+            'warmup_rate',
+            'update_freq',
+            'fp16',
+            'max_train_time',
+            'no_cuda',
+        ],
+    ),
+    'pytorch info': ('pytorch', []),
+}
+
+
+USER_SYM_SECTION = 'user_included^-^:'
+
+# using a list for easier insertion + needs to be ordered.
+section_list = [
+    'model_details',
+    'model_details:_quick_usage',
+    'model_details:_sample_input_and_output',
+    USER_SYM_SECTION + 'intended_use',
+    USER_SYM_SECTION + 'limitations',
+    USER_SYM_SECTION + 'privacy',
+    'datasets_used',
+    'evaluation',
+    'extra_analysis',
+    USER_SYM_SECTION + 'related_paper',
+    'hyperparameters',
+    'feedback',
+]
+
+# sections that have unique functions
+defined_sections = {
+    'model_details',
+    'quick_usage',
+    'sample_input_and_output',
+    'datasets_used',
+    'evaluation',
+    'extra_analysis',
+    'quantitative_analyses',
+    'safety_benchmark',
+    'feedback',
+    'hyperparameters',
+}
+
+# default messages for certain sections
+defaults = {
+    "intended_use": "This model is intended for the use of....\t",
+    "privacy": "This model has the following privacy concerns....\t",
+    "limitations": "This model has has these limitations: ...\t",
+}
+
+# special sections that either have...
+special_section = {
+    # different than expected headings
+    'related_paper': "Related Paper(s)",
+    'evaluation': 'Evaluation Results',
+    # don't include titles
+    'extra_analysis': None,
+}
+
+# types of functions; I get the strings mixed ups easily
+CLASSIFIER = 'classifier'
+GENERATOR = 'generator'
+RETRIEVER = 'retriever'
+RANKER = 'ranker'
+
+# different modes
+M_gen = 'gen'
+M_edit = 'editing'
+M_final = 'final'
+
+# dictionary of all models with their path as the key
+all_models = {model['path']: model for model in model_list}
+# dictionary of all tasks with their task field as the key
+all_tasks = {task['task']: task for task in task_list}
+
+# set containing internal tasks where we can just remove "internal:" and it will be okay
+internal_remove_tasks = {
+    "internal:blended_skill_talk",
+    "internal:light_dialog",
+    "internal:light_dialog",
+    "internal:eli5",
+    "internal:igc",
+    "internal:wizard_of_internet",
+}
+
+# dictionary mapping original task to new location, if they exist
+remap_task = {
+    "internal:safety:wikiToxicComments": "dialogue_safety:wikiToxicComments",
+    "internal:safety:adversarial": "dialogue_safety:adversarial",
+    "internal:personal_knowledge:PersonalTopicFollowup": "msc:PersonaSummary",
+    # TODO: check if this is actually correct
+    "internal:safety:multiturnConvAI2": "dialogue_safety:multiturn",
+    "internal:safety:boring": None,
+    "internal:convai2_review": None,
+    "internal:safety:boringConvAI2Review": None,
+    "internal:comment_battle:ImageDialogGenerationTeacher": None,
+    "internal:safety:adversarialConvAI2Review": None,
+    "internal:new_reddit:small": None,
+}
+
```

Comment:
added `internal_remove_tasks` and `remap_task` so that it's easier to deal with internal datasets

Reply:
I'd suggest we move the internal_remove_tasks and remap_task to internal igmc. This is confusing to all public ParlAI users (as they have no idea what it means?)